build/yaml/deployBotResources/js/deploy.yml

```diff
@@ -135,66 +141,76 @@ stages:
                 Write-Host "`nUsing the registry source: $registry";
                 New-Item -Path . -Name ".npmrc" -ItemType "file" -Value "registry=$registry";
 
-          # Install Packages
-          - task: Npm@1
-            displayName: 'Install dependencies'
-            inputs:
-              command: 'install'
-              workingDir: '$(SYSTEM.DEFAULTWORKINGDIRECTORY)/${{ bot.project.directory }}'
-              customRegistries: 'useNpmrc'
-              verbose: true
-
-          # Prepate bot
           - task: AzureCLI@2
-            displayName: 'Prepare Bot'
+            displayName: "Build and Push Container"
             inputs:
               azureSubscription: "${{ parameters.azureSubscription }}"
+              addSpnToEnvironment: true
+              failOnStderr: true
               scriptType: pscore
               scriptLocation: inlineScript
-              inlineScript: 'az bot prepare-deploy --code-dir "${{ bot.project.directory }}" --lang Javascript'
+              workingDirectory: $(SYSTEM.DEFAULTWORKINGDIRECTORY)/${{ bot.project.directory }}
+              inlineScript: |
+                az acr login --name ${{ bot.container.registry }} --username="$env:SERVICEPRINCIPALID" --password="$env:SERVICEPRINCIPALKEY"
```

Comment:
Shouldn't we pass these SERVICEPRINCIPALID and DEPENDENCIESVERSIONNUMBER as parameters?

Reply:
These two variables are only available to this task thanks to the `addSpnToEnvironment: true` input, adding them as parameter and passing `$env:SERVICEPRINCIPALID` for example will be the same, because the parameter will contain, not the value but the key `$env:SERVICEPRINCIPALID` and when it's resolved inside the script it will output `$env:SERVICEPRINCIPALID` and the script will know how interpret that.